### PR TITLE
Add local ephemeral storage support in LimitRange

### PR DESCRIPTION
**Pull request by NickrenREN**
**Special notes for your reviewer**:
For a single local ephemeral storage resource

xref #43607 

**Release note**:
```release-note
Add local ephemeral storage support to LimitRange
```

/assign @jingxu97 
cc @ddysher

### CODE REVIEW

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -3459,6 +3459,11 @@ func validateLimitRangeTypeName(value string, fldPath *field.Path) field.ErrorLi
 // Validate limit range resource name
 // limit types (other than Pod/Container) could contain storage not just cpu or memory
 func validateLimitRangeResourceName(limitType api.LimitType, value string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if value == string(api.ResourceEphemeralStorage) &&
+		!utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
+		return append(allErrs, field.Forbidden(fldPath, "ResourceEphemeralStorage field disabled by feature-gate for Resource LimitRange"))
```
Comment (jingxu97) [inline pkg/api/validation/validation.go]: indent here

File: pkg/api/validation/validation_test.go [hunk 2]
```diff
@@ -8609,6 +8609,71 @@ func getStorageResourceList(storage string) api.ResourceList {
...
+func getLocalStorageResourceList(ephemeralStorage string) api.ResourceList {
+	res := api.ResourceList{}
+	if ephemeralStorage != "" {
+		res[api.ResourceEphemeralStorage] = resource.MustParse(ephemeralStorage)
+	}
+	return res
+}
+
+func TestValidateLimitRangeForLocalStorage(t *testing.T) {
+	successCases := []struct {
```
Comment (ddysher) [inline pkg/api/validation/validation_test.go]: Is there errorCases?

File: pkg/api/validation/validation_test.go [hunk 3]
```diff
@@ -8609,6 +8609,71 @@ func getStorageResourceList(storage string) api.ResourceList {
 	return res
 }
 
+func getLocalStorageResourceList(ephemeralStorage string) api.ResourceList {
```
Comment (ddysher) [inline pkg/api/validation/validation_test.go]: Can u reuse `getLocalStorageResourceList` in validation_test.go and admission_test.go ?

Comment (NickrenREN) [reply pkg/api/validation/validation_test.go]: I wrote them in two files just like `getStorageResourceList`, and was not sure whether we need to put them into a utility package.

File: pkg/api/validation/validation_test.go (hunk 2 above)
Comment (NickrenREN) [reply pkg/api/validation/validation_test.go]: No, i copied it from `TestValidateLimitRange`, and intended to test local ephemeral storage feature gate here

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (jingxu97) [reply pkg/api/validation/validation.go]: ping here

**Comment by jingxu97** [comment]
cc @davidopp

File: pkg/api/validation/validation_test.go (hunk 2 above)
Comment (NickrenREN) [reply pkg/api/validation/validation_test.go]: Modified cases name

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (NickrenREN) [reply pkg/api/validation/validation.go]: Sorry , i am not sure what i should do here. i have already run `gofmt` here.
Make `if` condition in one line.

**Comment by thockin** [comment]
Is there no issue with which this is associated?

**Comment by jingxu97** [comment]
@thockin there is no issue. This belongs to local ephemeral storage isolation feature #43607.

**Comment by thockin** [comment]
So there IS an issue.

**Pull request merged**